File: tests/fixtures/factory-neg/impl.js
// WHEN NOT TO APPLY: exactly one concrete type, constructed once, no variation.
// A Factory would add indirection with zero payoff.
class Clock {
  constructor() { this.kind = "system"; }
  now() { return 42; }
}

const clock = new Clock();

function currentTick() {
  return clock.now();
}

module.exports = { currentTick };

File: tests/fixtures/py-factory-neg/impl.py
# WHEN NOT TO APPLY: exactly one concrete type, constructed once, no variation.
# A Factory would add indirection with zero payoff.


class Clock:
    kind = "system"

    def now(self) -> int:
        return 42


clock = Clock()


def current_tick() -> int:
    return clock.now()
